skcosmo/feature_selection/pcov_fps.py

```diff
@@ -0,0 +1,136 @@
+            :math:`{\\alpha}`
+
+    n_features_to_select : int or float, default=None
+        The number of features to select. If `None`, half of the features are
+        selected. If integer, the parameter is the absolute number of features
+        to select. If float between 0 and 1, it is the fraction of features to
+        select.
+
+    initialize: int or 'random', default=0
+        Index of the first feature to be selected. If 'random', picks a random
+        value when fit starts.
+
+    progress_bar: boolean, default=False
+                  option to use `tqdm <https://tqdm.github.io/>`_
+                  progress bar to monitor selections
+
+    tolerance: float, defaults to 1E-12
+                threshold below which distances will be considered 0
+
+    Attributes
+    ----------
+    haussdorf_ : ndarray of shape (n_features,)
+                 the minimum distance from each feature to the set of selected
+                 features in PCovR space. once a feature is selected, the
+                 distance is not updated; the final list will reflect the
+                 distances when selected.
+
+    norms_ : ndarray of shape (n_features,)
+        The self-covariances of each of the features
+
+    n_features_to_select : int
+        The number of features that were selected.
+
+    X_selected_ : ndarray (n_samples, n_features_to_select)
+                  The features selected
+
+    eligible_ : ndarray of shape (n_features,), dtype=bool
+        A mask of features eligible for selection
+
+    n_selected_ : int
+        The number of features that have been selected thus far
+
+    report_progress : callable
+        A wrapper to report the progress of the selector using a `tqdm` style
+        progress bar
+
+    score_threshold : float (optional)
+        A score below which to stop selecting points
+
+    selected_idx_ : ndarray of integers
+                    indices of the selected features, with respect to the
+                    original fitted matrix
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        mixing=0.5,
+        n_features_to_select=None,
+        initialize=0,
+        tolerance=1e-12,
+        progress_bar=False,
+    ):
+
+        if mixing == 1.0:
+            raise ValueError(
+                "Mixing = 1.0 corresponds to traditional FPS."
+                "Please use the FPS class."
+            )
+
+        self.mixing = mixing
+
+        super().__init__(
+            initialize=initialize,
+            n_features_to_select=n_features_to_select,
+            progress_bar=progress_bar,
+            tolerance=tolerance,
+        )
+
+    def _set_norms(self, X, y):
+        """
+        Set the norms as the augmented dot product of features
+        """
+        self.modified_covariance_ = pcovr_covariance(mixing=self.mixing, X=X, Y=y)
```

Comment:
It's a bit out of the scope of this PR, but we should probably be using lowercase "y" in all of our PCovR-based methods to be consistent with sklearn.